Integrations/integration-email-sender.yml

```diff
@@ -0,0 +1,349 @@
+
+
+    # Following methods raise exceptions so no need to check for return codes
+    SERVER = SMTP(demisto.getParam('host'), int(demisto.getParam('port')))
+    SERVER.ehlo()
+    # TODO - support for non-valid certs
+    if demisto.getParam('tls'):
+        SERVER.starttls()
+    if demisto.getParam('credentials') and demisto.getParam('credentials')['identifier'] and demisto.getParam('credentials')['password']:
+        SERVER.login(demisto.getParam('credentials')['identifier'], demisto.getParam('credentials')['password'])
+    FROM = demisto.getParam('from')
+
+
+    def randomword(length):
+        """
+        Generate a random string of given length
+        """
+        letters = string.ascii_lowercase
+        return ''.join(random.choice(letters) for i in range(length))
+
+
+    def return_error(data):
+        """
+        Return error as result and exit
+        """
+        SERVER.quit()
+        demisto.results({
+            'Type': entryTypes['error'],
+            'ContentsFormat': formats['text'],
+            'Contents': data
+        })
+        sys.exit(0)
+
+
+    def guess_type(filename):
+        """
+        Return the maintype and subtype guessed based on the extension
+        """
+        ctype, encoding = mimetypes.guess_type(filename)
+        if ctype is None or encoding is not None:
+            # No guess could be made, or the file is encoded (compressed), so
+            # use a generic bag-of-bits type.
+            ctype = 'application/octet-stream'
+        return ctype.split('/', 1)
+
+
+    def handle_file(msg, filename, maintype, subtype, cid, data):
+        """
+        Add the attachment to the message and add the relevant header
+        """
+        if maintype == 'text':
+            # UTF-8 is a pretty safe bet
+            att = MIMEText(data, subtype, 'utf-8')
+        elif maintype == 'image':
+            att = MIMEImage(data, subtype)
+        elif maintype == 'audio':
+            att = MIMEAudio(data, subtype)
+        else:
+            att = MIMEBase(maintype, subtype)
+            att.set_payload(data)
+            # Encode the payload using Base64
+            encoders.encode_base64(att)
+        # Set the filename parameter
+        if cid:
+            att.add_header('Content-Disposition', 'inline', filename=filename)
+            att.add_header('Content-ID', '<' + cid + '>')
+        else:
+            att.add_header('Content-Disposition', 'attachment', filename=filename)
+        msg.attach(att)
+
+
+    def handle_html(htmlBody):
+        """
+        Extract all data-url content from within the html and return as separate attachments.
+        Due to security implications, we support only images here
+        We might not have Beautiful Soup so just do regex search
+        """
+        attachments = []
+        cleanBody = ''
+        lastIndex = 0
+        for i, m in enumerate(re.finditer(r'<img.+?src=\"(data:(image\/.+?);base64,([a-zA-Z0-9+/=]+?))\"', htmlBody, re.I)):
```

Comment:
Do you want to optimize for cases the file will appear more than once in the html
logos and such ...

Reply:
Nope - good enough as is...